File: api/booking-service/src/main/java/com/carrental/bookingservice/model/dto/LocationAddDTO.java
package com.carrental.bookingservice.model.dto;

import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.Size;

@Data
public class LocationAddDTO {

    @NotEmpty
    @Size(max = 50)
    private String country;

    @NotEmpty
    @Size(max = 150)
    private String city;

    @NotEmpty
    @Size(max = 150)
    private String streetAndNb;

    @NotEmpty
    @Size(max = 20)
    private String code;

    @NotEmpty
    @Size(max = 255)
    private String email;

    @NotEmpty
    @Size(max = 30)
    private String phone;
}

File: frontend-app/shared/model/LocationAddDTO.ts
export class LocationAddDTO {
    country: string;
    city: string;
    streetAndNb: string;
    code: string;
    email: string;
    phone: string;

    constructor(country: string, city: string, streetAndNb: string, code: string, email: string, phone: string) {
        this.country = country;
        this.city = city;
        this.streetAndNb = streetAndNb;
        this.code = code;
        this.email = email;
        this.phone = phone;
    }
}
